frontend/src/components/account/ContactDetails/ContactDetails.jsx

```diff
@@ -1,14 +1,25 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import styles from './ContactDetails.module.scss';
 import { ICONS } from '../../ui-kit/icons';
 import Button from '../../ui-kit/components/Button/Button';
 import { useNavigate } from 'react-router-dom';
 import Account from '../Account';
-import { useSelector } from 'react-redux';
+import { useDispatch } from 'react-redux';
+import { login } from '../../../redux/slices/userSlice';
 
 const ContactDetails = () => {
+  const dispatch = useDispatch();
   const navigate = useNavigate();
-  const userData = useSelector((state) => state.user.user);
+  const storedUserData = JSON.parse(localStorage.getItem('userfetchedData'));
+  const [userData, setUserData] = useState(storedUserData);
+
+  useEffect(() => {
+    const storedUserData = JSON.parse(localStorage.getItem('userfetchedData'));
```

Comment:
I'm not sure you need this useEffect. You already took information from local storage, and it will be done on each render. What is the point in useEffect?

Reply:
useEffect is to update data from local storage when there is a change in the Redux store. Without the useEffect, it will be fetching the data from local storage on every render.

Reply:
Well, it doesn't work that way. On the initial render it takes data from local storage, after that you cause rerendering with useEffect. And the component renders every time we visit the profile page, so there is no need for useEffect.

Reply:
what about updating info when user changes it?

Reply:
As far as I understand, users can only change their personal data in the User Cabinet. There you have two components: one for displaying info, one for changing it. After user saves the info, first component renders again and gets new info. I just tried it without useEffect and it works fine for me.